src/device-registry/controllers/manage-site.js

```diff
@@ -22,51 +23,91 @@ const {
 const getDetail = require("../utils/get-device-details");
 
 const manageSite = {
-  doActivity: async (req, res) => {
-    try {
-      const { type, tenant } = req.query;
-      if (tenant && type) {
-        const { deviceName } = req.body;
-
-        const device = await getDetail(tenant, deviceName);
-        const doesDeviceExist = !isEmpty(device);
-        const isDeployed = await isDeviceDeployed(
-          deviceName,
-          tenant.toLowerCase()
-        );
-        const isRecalled = await isDeviceRecalled(
-          deviceName,
-          tenant.toLowerCase()
-        );
-        const { siteActivityBody, deviceBody } = siteActivityRequestBodies(
-          req,
-          res
-        );
-        logElement("does the device exist", doesDeviceExist);
-        logElement("is the device deployed", isDeployed);
-
-        doLocationActivity(
-          res,
-          deviceBody,
-          siteActivityBody,
-          deviceName,
-          type,
-          doesDeviceExist,
-          isDeployed,
-          isRecalled,
-          tenant.toLowerCase()
-        );
-      } else {
-        return res.status(HTTPStatus.BAD_REQUEST).json({
-          success: false,
-          message: "missing query params, please check documentation",
-        });
-      }
-    } catch (e) {
-      tryCatchErrors(res, e);
+  recallDevice: async  (req, res) => {
+    const { tenant, deviceName } = req.query;
+    const isRecalled = await isDeviceRecalled(
+        deviceName,
+        tenant.toLowerCase()
+    );
+    if (isRecalled) {
+      return res.status(HTTPStatus.CONFLICT).json({
+      success: false,
+      message: `Device ${deviceName} already recalled`,
+    })
+    }
+    const { siteActivityBody, deviceBody } = siteActivityRequestBodies(req, res, "recall");
+    return await carryOutActivity(
+        res,
+        tenant,
+        deviceName,
+        deviceBody,
+        siteActivityBody,
+        {
+          successMsg: `Successfully recalled device ${deviceName}`,
+          errorMsg: `Failed to recall device ${deviceName}`,
+        }
+    )
+
+  },
+  deploymentFields: [
+      "height",
+      "mountType",
+      "powerType",
+      "date",
+      "latitude",
+      "longitude",
+      "isPrimaryInLocation",
+      "isUserForCollocaton"
```

Comment:
was this supposed to be isUsedForCollocation. just a check

Reply:
@Baalmart @sserurich is has been fixed, thanks.